src/bucos.erl

```diff
@@ -10,7 +10,7 @@ run(Cmd) ->
 % @doc
 % Execute the given shell command
 % @end
--spec run(string() | binary(), integer()) -> {ok, term()} | {error, term()}.
+-spec run(string() | binary(), integer()) -> {ok, term()} | {error, term()} | {error, term(), string()}.
```

Comment:
You break the backward compatibility. Maybe you need to create a new function : 
```
-spec run(string() | binary(), [options()]) -> {ok, term()} | {error, term()} | {error, term(), string()}.
```

Where `options()` is a list with the following options :

* `{timeout, integer()} : for timeout
* `stdout_on_error` : to return `{error, integer(), string()}` on error